dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/SyntacticContextAnalysis.java

```diff
@@ -100,9 +99,6 @@ public String toString() {
         }
     }
 
-    // We use this enum to track loop nesting
-    private enum LoopMarkerTypes {START, INC, END}
-
```

Comment:
Where are the changes in this class coming from? Are they really related to this PR or is it just a "complementary fix"? Maybe you noticed this now that you print the alias graph.

I remember seeing some problems related to this when generating some FAIL report, but I never got the time to investigate it further.

Reply:
Yes, these problems occured when using `SyntacticContextAnalysis` to debug problems.
The problem was that if you had an empty iteration body (say due to propagating the whole loop body via CP),
then the iteration entry label was both beginning and end of an iteration.
The analysis didn't understand that an event could be both and so it ended up treating it as "end", which caused an exception because there was no matching "begin".

I have not checked if the changes by René are a proper fix (though I discussed this issue with him). Either way, the changes are complementary and unrelated to the alias analysis itself.

Reply:
My changes assume that iterations of the same loop do not overlap and that nested loops do not touch the boundaries of their outer loop.  This should hold according to `LoopAnalysis`.  However, events that were placed in between iterations are no longer recognized as being inside the loop.

Reply:
> However, events that were placed in between iterations are no longer recognized as being inside the loop

Doesn't this mean that the analysis is even more broken now? Or do I misunderstand something?